haskell/cabal.bzl

```diff
@@ -478,15 +485,20 @@ def _compute_dependency_graph(repository_ctx, versioned_packages, unversioned_pa
     if not versioned_packages and not unversioned_packages:
         return ({}, [])
 
-    # Unpack all given packages, then compute the transitive closure
-    # and unpack anything in the transitive closure as well.
     stack_cmd = repository_ctx.which("stack")
     if not stack_cmd:
         fail("Cannot find stack command in your PATH.")
-    stack = [stack_cmd, "--no-nix", "--skip-ghc-check", "--system-ghc"]
+    exec_result = _execute_or_fail_loudly(repository_ctx, [stack_cmd, "--version"])
+    stack_major_version = int(exec_result.stdout.split(" ")[0].split(".")[0])
+    if stack_major_version < 2:
+        fail("Stack 2.1 or above required.")
+
+    # Unpack all given packages, then compute the transitive closure
+    # and unpack anything in the transitive closure as well.
+    stack = [stack_cmd]
     if versioned_packages:
         _execute_or_fail_loudly(repository_ctx, stack + ["unpack"] + versioned_packages)
-    stack = [stack_cmd, "--no-nix", "--skip-ghc-check", "--system-ghc", "--resolver", repository_ctx.attr.snapshot]
+    stack = [stack_cmd, "--resolver", repository_ctx.attr.snapshot]
```

Comment:
Why are those flags no longer necessary? Or have they been removed?

Reply:
These flags were meant to prevent Stack from trying to futz around with GHC. Newer Stack is now more lazy about pulling in GHC, so these flags are no longer required,  in particular with `--global-hints`.